projects/core/components/button/button.component.ts

```diff
@@ -107,7 +107,7 @@ export class TuiButtonComponent
 
     @HostBinding(`attr.data-appearance`)
     get computedAppearance(): string {
-        return this.appearance ?? (this.options.appearance || ``);
+        return this.appearance || this.options.appearance || ``;
```

Comment:
No, there's gotta be `null` reintroduced to `TuiAppearance`. Otherwise `appearance=""` does not remove appearance but instead turns it back to the default.